### PR TITLE
[GLUTEN-4917][VL] GHA with pre-built docker image and github runner

### PR DESCRIPTION
**Pull request by zhouyuan**
## What changes were proposed in this pull request?

allow to use pre-built docker image in GHA
running in github runner

gap:
- cpp benchmark, HBM tests
- celeborn tests
- oom/randome kill tests
- iceberg/delta lake tests

## How was this patch tested?

pass GHA

### CODE REVIEW

File: .github/workflows/velox_docker.yml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      - uses: actions/checkout@v2
+      - name: Build Gluten velox third party
+        run: |
+          yum install sudo patch java-1.8.0-openjdk-devel -y && \
+          cd $GITHUB_WORKSPACE/ep/build-velox/src && \
+          ./get_velox.sh && \
+          source /opt/rh/devtoolset-9/enable && \
+          source $GITHUB_WORKSPACE//dev/vcpkg/env.sh && \
+          cd $GITHUB_WORKSPACE/ && \
+          sed -i '/^headers/d' ep/build-velox/build/velox_ep/CMakeLists.txt && \
```
Comment (zhouyuan) [inline .github/workflows/velox_docker.yml:59]: to use the boost installed in vcpkg

**Comment by zhouyuan** [comment]
@PHILO-HE

File: gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 2]
```diff
@@ -71,7 +71,7 @@ class VeloxTestSettings extends BackendTestSettings {
   enableSuite[GlutenWriterColumnarRulesSuite]
 
   enableSuite[GlutenQueryCompilationErrorsDSv2Suite]
-  enableSuite[GlutenQueryCompilationErrorsSuite]
```
Comment (zhouyuan) [inline gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:74]: The suite seems buggy on new runner, but only for Spark33. Looks odd, so I disabled it first
```
2024-03-20T05:47:06.2483725Z [31m- UNSUPPORTED_FEATURE: Using Python UDF with unsupported join condition *** FAILED ***[0m
2024-03-20T05:47:06.2486680Z [31m  Expected exception org.apache.spark.sql.AnalysisException to be thrown, but java.lang.RuntimeException was thrown (QueryCompilationErrorsSuite.scala:144)[0m
```

File: .github/workflows/velox_docker.yml [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+          cd $GITHUB_WORKSPACE/tools/gluten-it
+          export MAVEN_HOME=/usr/lib/maven
+          export PATH=${PATH}:${MAVEN_HOME}/bin
+          mvn clean install -P${{ matrix.spark }} \
+          && GLUTEN_IT_JVM_ARGS=-Xmx5G sbin/gluten-it.sh queries-compare \
+            --local --preset=velox --benchmark-type=h --error-on-memleak --off-heap-size=10g -s=1.0 --threads=16 --iterations=1 \
+          && GLUTEN_IT_JVM_ARGS=-Xmx5G sbin/gluten-it.sh queries-compare \
+            --local --preset=velox --benchmark-type=ds --error-on-memleak --off-heap-size=10g -s=1.0 --threads=16 --iterations=1
+     
+  # run-tpc-test-centos8-oom-randomkill:
```
Comment (zhouyuan) [inline .github/workflows/velox_docker.yml:184]: reports OOM on current GH runner, need to run with proper config

File: .github/workflows/velox_docker.yml [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+          tar --strip-components=1 -xf v3.3.1.tar.gz spark-3.3.1/sql/core/src/test/resources/  && \
+          mkdir -p shims/spark33/spark_home/ && \
+          mv sql shims/spark33/spark_home/ 
+      - name: Build and Run unit test for Spark 3.3.1 (other tests)
+        run: |
+          cd $GITHUB_WORKSPACE/ && \
+          export SPARK_SCALA_VERSION=2.12 && \
+          export MAVEN_HOME=/usr/lib/maven 
+          export PATH=${PATH}:${MAVEN_HOME}/bin
+          mvn clean install -Pspark-3.3 -Pbackends-velox -Prss -Pspark-ut -DargLine="-Dspark.test.home=$GITHUB_WORKSPACE//shims/spark33/spark_home/" -DtagsToExclude=org.apache.spark.tags.ExtendedSQLTest,io.glutenproject.tags.UDFTest,io.glutenproject.tags.SkipTestTags && \
```
Comment (zhouyuan) [inline .github/workflows/velox_docker.yml:455]: disabled iceberg test due to some version-hint.txt missing
disabled delta tests due to heap OOM

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Comment by zhouyuan** [comment]
Hi @ulysses-you @zwangsheng 
we just find the the golden file tests are not covered, could you please help to re-enable this, if this is still required? 

thanks, 
-yuan

**Comment by ulysses-you** [comment]
Is there any block on enabling golden files test ? If not, I can help to recover it.

**Comment by zhouyuan** [comment]
> Is there any block on enabling golden files test ? If not, I can help to recover it.


https://github.com/apache/incubator-gluten/blob/main/.github/workflows/velox_be.yml.deprecated#L104

If I understand correctly, we may just need to upload the artifact now,  the "docker cp" is not required

thanks,
-yuan

**Pull request merged**